katsdpdata/met_handler.py

```diff
@@ -33,6 +33,111 @@ def __init__(self, solr_url, product_type, product_name, product_id=None):
         else:
             self.product_id = str(uuid.uuid4())
 
+    def get_prod_met(self, prod_id=None):
+        """Get the metadata for a product, if prod_id is not specified, use self.prod_id.
+
+        Parameters
+        ----------
+        prod_id: string : the product id to find.
+
+        Returns
+        -------
+        met: dict : metadata containing the _version_ for version tracking commits to solr.
+        """
+        copy_fields = ['Observer_lowercase']
+        if not prod_id:
+            prod_id = self.prod_id
+        query = 'id:{}'.format(prod_id)
+        res = self.solr.search(query)
+        if res.hits == 0:
+            return None
+        elif res.hits > 1:
+            raise MetaDataHandlerException('{} returned for {}'.format(res.hits, query))
+        # filter out copy field
+        doc = res.docs[0]
+        for c in copy_fields:
+            doc.pop(c, None)
+        return doc
+
+    def del_prod_met(self, prod_id):
+        """Delete product metadata. Use with caution, as will delete product metadata
+        from solr.
+
+        Parameters
+        ----------
+        prod_id: string : the product id to delete.
+        """
+        met = self.get_prod_met(prod_id)
+        self.solr.delete(id=prod_id)
+        return met
+
+    def set_product_transferring(self, met):
+        """Set product transfer status while transfering to the backend storage.
+
+        Parameters
+        ----------
+        met: dict : metadata dict, a local copy of the solr doc to update.
+
+        Returns
+        -------
+        met: dict : metadata containing the _version_ for version tracking commits to solr.
+        """
+        met['CAS.ProductReceivedTime'] = ''
```

Comment:
No point storing an empty string. This will still be indexed by solr, so this will increase the insertion time and take up space. You can search for empty fields in solr, but at least they don't get indexed.

Reply:
This assumes the ProductReceivedTime is only set once. I guess if the ProductRecievedTime is already set you could set it to an empty string, but I'm not sure an empty string would validate. Might be better to delete the product and reset it without the revcieved time.